rdflib/plugins/parsers/notation3.py

```diff
@@ -349,7 +349,7 @@ def unicodeExpand(m):
 signed_integer = re.compile(r'[-+]?[0-9]+')      # integer
 integer_syntax = re.compile(r'[-+]?[0-9]+')
 decimal_syntax = re.compile(r'[-+]?[0-9]*\.[0-9]+')
-exponent_syntax = re.compile(r'[-+]?(?:[0-9]+\.[0-9]*(?:e|E)[-+]?[0-9]+|'+
+exponent_syntax = re.compile(r'[-+]?(?:[0-9]*\.[0-9]*(?:e|E)[-+]?[0-9]+|'+
                              r'\.[0-9](?:e|E)[-+]?[0-9]+|'+
```

Comment:
Your fix will now match ".e1".

What it looks like this was trying to do is allow ".1e1", "1.e1" and "1e1", which means you need `[0-9]*\.[0-9]+`, `[0-9]+\.[0-9]*` and `[0-9]+`. The problem is line 2 is missing a `+`:

```suggestion
exponent_syntax = re.compile(r'[-+]?(?:[0-9]+\.[0-9]*(?:e|E)[-+]?[0-9]+|'+
                             r'\.[0-9]+(?:e|E)[-+]?[0-9]+|'+
```